File: src/cpp/cdm/compartment/substances/SESubstanceQuantity.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "cdm/CommonDefs.h"

#include "cdm/compartment/substances/SESubstanceQuantity.h"
#include "cdm/substance/SESubstance.h"

SESubstanceQuantity::SESubstanceQuantity(SESubstance& sub) : Loggable(sub.GetLogger()), m_Substance(sub)
{

};
SESubstanceQuantity::~SESubstanceQuantity()
{

};

File: src/java/com/kitware/pulse/cdm/substance/quantity/SESubstanceQuantity.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package com.kitware.pulse.cdm.substance.quantity;

import com.kitware.pulse.cdm.bind.SubstanceQuantity.SubstanceQuantityData;
import com.kitware.pulse.cdm.substance.SESubstance;

public abstract class SESubstanceQuantity 
{
  protected SESubstance           substance;
  
  public SESubstanceQuantity(SESubstance s)
  {
    substance=s;
  }
  public void clear()
  {
    
  }
  public static void load(SubstanceQuantityData src, SESubstanceQuantity dst)
  {
    dst.clear();
  }
  protected static void unload(SESubstanceQuantity src, SubstanceQuantityData.Builder dst)
  {
    if(src.substance!=null)
      dst.setSubstance(src.getSubstance().getName());
  }
  
  public SESubstance getSubstance()
  {
    return substance;
  }
}
